### PR TITLE
PR on top of PR: kiz frame api

### PR DESCRIPTION
**Pull request by gui1117**
on top of https://github.com/paritytech/substrate/pull/14137
cc @kianenigma 

with this PR pallet-example-basic compiles without frame_benchmarking, frame_system and frame_support dependency.

Code is not well written, but this is enough for ensuring that it works

For the construct_runtime/tt_default_parts, I reverted the change so that path to `frame_support` to passed along.
For the benchmarking macro, only system crate needed to be abstracted

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014464

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014471

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014548

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014549

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014550

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3014555

**Review by sam0x17 — APPROVED** [review]
lgtm!

File: frame/examples/basic/Cargo.toml [hunk 1]
```diff
@@ -16,9 +16,7 @@ targets = ["x86_64-unknown-linux-gnu"]
 codec = { package = "parity-scale-codec", version = "3.2.2", default-features = false }
 scale-info = { version = "2.5.0", default-features = false, features = ["derive"] }
 log = { version = "0.4.17", default-features = false }
-frame-benchmarking = { version = "4.0.0-dev", default-features = false, optional = true, path = "../../benchmarking" }
-frame-support = { version = "4.0.0-dev", default-features = false, path = "../../support" }
-frame-system = { version = "4.0.0-dev", default-features = false, path = "../../system" }
+frame = { version = "0.0.1-dev", default-features = false, path = "../../../frame/" }
```
Comment (kianenigma) [inline frame/examples/basic/Cargo.toml:19]: If we are coming this far, should we refactor the whole example to also not use sp-* as well? everything should be re-exported from frame. 

Or else, we can leave pallet-example-basic as is for now, and add benchmarking to pallet-example-frame-api.

File: frame/examples/basic/src/benchmarking.rs [hunk 2]
```diff
@@ -21,8 +21,8 @@
 #![cfg(feature = "runtime-benchmarks")]
 
 use crate::*;
-use frame_benchmarking::v2::*;
-use frame_system::RawOrigin;
+use frame::deps::frame_benchmarking::v2::*;
```
Comment (kianenigma) [inline frame/examples/basic/src/benchmarking.rs:24]: would be good to have a benchmarking_prelude in frame crate now.

**Review by kianenigma — APPROVED** [review]
have some further suggestions, bit happy to merge as is for improving tt_call stuff, and making frame benchmarking macros work with frame-api.

File: frame/examples/basic/src/benchmarking.rs (hunk 2 above)
Comment (thiolliere) [reply frame/examples/basic/src/benchmarking.rs:24]: Yes I agree

File: frame/examples/basic/Cargo.toml (hunk 1 above)
Comment (thiolliere) [reply frame/examples/basic/Cargo.toml:19]: > Or else, we can leave pallet-example-basic as is for now, and add benchmarking to pallet-example-frame-api.

maybe it is better to leave pallet-example-basic as is for now, I used it only as a tests, if we move basic example to frame we shouldn't use the re-export `frame::deps` so much also, it needs to be cleaned

Comment (kianenigma) [reply frame/examples/basic/Cargo.toml:19]: Agreed, then let's do all experiments in `pallet-example-frame-api` for now.

**Comment by thiolliere** [comment]
this PR was open as an example

**Pull request closed without merging**